project/src/Systems/GameManager.gd

```diff
@@ -0,0 +1,22 @@
+extends Node
+
+
+onready var world: Node2D = $World
+onready var player: Node = $World/PlayerShip
+onready var pirate_objects := $World/Pirates
+onready var camera := $World/Camera
+onready var world_objects: = get_tree().get_nodes_in_group("WorldObjects")
+
+
+func _ready() -> void:
+	player.connect("player_dead", self, "_on_Player_dead")
+	for po in pirate_objects.get_children():
+		po.setup_target(player.agent)
+		po.setup_world_objects(world_objects)
+	camera.setup_player(player)
+
+
+func _on_Player_dead() -> void:
+	for po in pirate_objects.get_children():
```

Comment:
How about connecting the pirateship to a `died` signal on the player and the camera? I'm not sure what's best here, but this seems like something that can change or break to me. I.e. the way the camera, enemies, etc. work.